oak_remote_attestation_verification/src/lib.rs

```diff
@@ -18,8 +18,23 @@
 
 extern crate alloc;
 
+pub mod proto {
+    pub mod oak {
+        tonic::include_proto!("oak");
+        pub mod attestation {
+            pub mod v1 {
+                tonic::include_proto!("oak.attestation.v1");
+            }
+        }
+    }
+}
+
+pub mod endorsement;
 pub mod rekor;
-pub mod verifier;
+pub mod verify;
+
+#[cfg(test)]
+mod endorsement_tests;
```

Comment:
The usual/idiomatic Rust approach for module-specific unit tests is to either add the tests directly to the module, or create a directory for the module which then contains a mod.rs and corresponding tests.rs. See e.g. https://github.com/project-oak/oak/blob/main/oak_virtio/src/queue/ or https://github.com/project-oak/oak/blob/b71e3e74bec46e56aba8918d0f36922e17124cf3/oak_core/src/sync.rs#L85

Reply:
Acknowledged. I find both alternatives suboptimal/unappealing - shouldn't mix implementation codes for one (also, the vertical scrolling quickly becomes a nuisance, not possible to switch between two tabs since it is the same file), and a directory for just two files seems unnecessary. Anyways.

Reply:
Wait, maybe I misunderstood. To clarify, is the present filename constellation allowable, and it is just this declaration which needs to be moved to the module?

Reply:
The problem with the current approach is that these become integration tests rather than unit tests in Rust terms, since it won't have access to internal details of the modules.

If it is not needed to have access to internal details of the modules, then these should be added as actual integration tests, which according to Rust convention should be in a `tests` directory. (see e.g. https://doc.rust-lang.org/rust-by-example/testing/integration_testing.html)

Reply:
> Wait, maybe I misunderstood. To clarify, is the present filename constellation allowable, and it is just this declaration which needs to be moved to the module?

That would not make it unit tests. Since the tests are not children of the module it won't have access to the implementation details, so should still become integration tests if you are only going to test against the public API.

Reply:
Done, as discussed offline.